### PR TITLE
Fixes #3946 Whitescreen if az_person not installed.

### LINKED ISSUE #3944: Navigation updates
## Motivation
The desire to update the navigation comes from extensive research on usability and user interface.

### Describe the solution you'd like
1. The desktop navigation should have a real page at each level, not just a "folder" structure.
2. Websites with deeper navigation and more content create for a better SEO than smaller separate sites.
3. There is a desire to allow more space for content on deeper pages -- especially if they are "micro-site" style content.
4. The mobile navigation should allow for more content and take up less real estate on the page itself.

## Additional context

- [2024 Quickstart Navigation Google Drive Folder](https://drive.google.com/drive/folders/1tCZnan3uyYgauzXUsysoPTm6oOMzf5nt?usp=drive_link)
- [Quickstart Navigation Proposal](https://docs.google.com/presentation/d/1iyYVrfCRoagBzckHZ1P4Byt42eZQsngyS6Mn9vtKBD0/edit#slide=id.p)

### Open Questions
- What are the defaults?
- What happens to the other content within the left sidebar region?
- What happens to the right sidebar content?
- What are changeable versus non-changeable aspects?
- Do we want color scheme options? Is that chosen by the site owner?

### LINKED ISSUE #3946: Site crashes on log in if az_person not enabled.
## Problem/Motivation
My site crashes when logging in with the following error in the log and a white screen.

```
Drupal\Core\Entity\Query\  
                                                     QueryException:            
                                                     'field_az_netid' not       
                                                     found in                   
                                                     Drupal\Core\Entity\Query\  
                                                     Sql\Tables->ensureEntityT  
                                                     able() (line 369 of        
                                                     /code/web/core/lib/Drupal  
                                                     /Core/Entity/Query/Sql  
```

### Describe the bug
Code in az_core module looks for entities with the field_az_netid field, but that field does not exist if az_person is not installed.

### To Reproduce

You must install quickstart without the az_person module.
Then log in and see the white screen.

## Proposed resolution
We should check if the person module is installed before checking for the field, or move the field to az_core.

##  Screenshots
If applicable, add screenshots to help explain your problem. (Need a tool to share screenshots or screencasts as URLs? Try [monosnap](https://monosnap.com)

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Checking if az_person is enabled before counting on az_person fields.

### Release notes
```
This release fixes a critical issue affecting AZ Quickstart 2.12.0 sites that do not have the `az_person` module installed that causes a fatal error to occur when users log in to the site.
```

## Related issues

Closes #3946 

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My change requires release notes.

### CODE REVIEW

**Review by joeparsons — DISMISSED** [review]

File: modules/custom/az_core/src/AZUserToolbarLinkBuilder.php [hunk 1]
```diff
@@ -36,11 +44,14 @@ class AZUserToolbarLinkBuilder extends ToolbarLinkBuilder {
    *   The entity type manager.
    * @param \Drupal\externalauth\AuthmapInterface $authmap
    *   The authmap service.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $moduleHandler
+   *  The module handler.
```
Comment (joeparsons) [inline modules/custom/az_core/src/AZUserToolbarLinkBuilder.php:48]:
```suggestion
   *   The module handler.
```

**Comment by joeparsons** [comment]
The affected file was added in 2.12.x only as part of #3764.  If this issue also affects 2.11.x, we'll need a separate PR for that.

**Review by joeparsons — DISMISSED** [review]

**Comment by tadean** [comment]
I think to prevent other similar issues we may want to check if the field exists on the bundle, rather than checking if the module is enabled.

File: modules/custom/az_core/src/AZUserToolbarLinkBuilder.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 namespace Drupal\az_core;
 
 use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
```
Comment (trackleft) [inline modules/custom/az_core/src/AZUserToolbarLinkBuilder.php:6]:
```suggestion
```

File: modules/custom/az_person/src/AZUserToolbarLinkBuilder.php [hunk 3]
```diff
@@ -14,32 +15,47 @@
...
+   *   The entity field manager.
    * @param \Drupal\externalauth\AuthmapInterface $authmap
    *   The authmap service.
    */
-  public function __construct(AccountProxyInterface $account, EntityTypeManagerInterface $entityTypeManager, ?AuthmapInterface $authmap) {
+  public function __construct(
+    AccountProxyInterface $account,
+    EntityTypeManagerInterface $entityTypeManager,
+    EntityFieldManagerInterface $entityFieldManager,
+    ?AuthmapInterface $authmap
```
Comment (trackleft) [inline modules/custom/az_person/src/AZUserToolbarLinkBuilder.php:54]:
```suggestion
    ?AuthmapInterface $authmap,
```

File: modules/custom/az_core/az_core.services.yml [hunk 4]
```diff
@@ -25,12 +25,13 @@ services:
       - '@config_update.config_list'
       - '@module_handler'
   # Decorates the normal user module links.
-  az_core.toolbar_link_builder:
+  az_person.toolbar_link_builder:
```
Comment (joeparsons) [inline modules/custom/az_core/az_core.services.yml:28]: Does this need to change back?
```suggestion
  az_core.toolbar_link_builder:
```

File: modules/custom/az_core/src/AZUserToolbarLinkBuilder.php [hunk 5]
```diff
@@ -14,32 +15,47 @@
 class AZUserToolbarLinkBuilder extends ToolbarLinkBuilder {
 
   /**
-   * Drupal\externalauth\AuthmapInterface definition.
+   * Authmap service definition.
```
Comment (trackleft) [inline modules/custom/az_core/src/AZUserToolbarLinkBuilder.php:18]:
```suggestion
   * Drupal\externalauth\AuthmapInterface definition.
```

**Review by tadean — APPROVED** [review]
Seems to work in cases where `az_person` does not exist, and `field_az_netid` does not exist. Thanks @trackleft !

**Review by joeparsons — APPROVED** [review]

**Pull request merged**